module/zfs/vdev_queue.c

```diff
@@ -747,6 +747,9 @@ vdev_queue_io(zio_t *zio)
 	vdev_queue_t *vq = &zio->io_vd->vdev_queue;
 	zio_t *nio;
 
+	if (!zio->io_timestamp)
```

Comment:
this should probably be explicit about checking for zero: `if (zio->io_timestamp == 0)`